### ISSUE TITLE
Crash: SDWebImageDownloaderOperation.m, line 528

### ISSUE THREAD

**Issue by tspop** [rs/SDWebImage#2042]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | ios
 Platform Version        | 10.3.3
 SDWebImage Version      | 4.0.0
 Integration Method      | cocoapods
 Xcode Version           | Xcode 8.3
 Repro rate              | only once in production

### Issue Description and Steps

```1	libobjc.A.dylib	objc_retain + 4
2	SDWebImage	-[SDWebImageDownloaderOperation callCompletionBlocksWithImage:imageData:error:finished:] (SDWebImageDownloaderOperation.m:528)
3	SDWebImage	-[SDWebImageDownloaderOperation URLSession:task:didCompleteWithError:] (SDWebImageDownloaderOperation.m:449)
4	SDWebImage	-[SDWebImageDownloader URLSession:task:didCompleteWithError:] (SDWebImageDownloader.m:299)
5	CFNetwork	__51-[NSURLSession delegate_task:didCompleteWithError:]_block_invoke.185 + 74
6	Foundation	__NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 6
7	Foundation	-[NSBlockOperation main] + 144
8	Foundation	-[__NSOperationInternal _start:] + 768
9	Foundation	__NSOQSchedule_f + 188
10	libdispatch.dylib	_dispatch_queue_serial_drain + 852
11	libdispatch.dylib	_dispatch_queue_invoke + 884
12	libdispatch.dylib	_dispatch_root_queue_drain + 324
13	libdispatch.dylib	_dispatch_worker_thread3 + 104
14	libsystem_pthread.dylib	_pthread_wqthread + 1038
15	libsystem_pthread.dylib	start_wqthread + 6```


Sorry if duplicate.

**Comment by dreampiggy** [comment]
Duplicated for #1998 .....Maybe we can release 4.1.1 in October. Any contributor idea? @bpoplauschi @iwill.

**Comment by bpoplauschi** [comment]
Indeed, I think we can release in a couple of days.
